Shared/Samples/Show line of sight between geoelements/ShowLineOfSightBetweenGeoelementsView.swift

```diff
@@ -0,0 +1,345 @@
+            .background(.thinMaterial, ignoresSafeAreaEdges: .horizontal)
+        }
+        .task {
+            await model.addGraphics()
+        }
+        .toolbar {
+            ToolbarItemGroup(placement: .bottomBar) {
+                Button("Settings") {
+                    isPresented = true
+                }
+                .sheet(isPresented: $isPresented) {
+                    settingsSheet
+                }
+            }
+        }
+        .errorAlert(presentingError: $model.error)
+    }
+    
+    /// The settings configuration sheet for adjusting the observer's height.
+    private var settingsSheet: some View {
+        NavigationStack {
+            Form {
+                let heightRange = 20.0...70.0
+                let numberFormat = FloatingPointFormatStyle<Double>.number.precision(.fractionLength(0))
+                LabeledContent(
+                    "Observer Height",
+                    value: model.height,
+                    format: numberFormat
+                )
+                Slider(
+                    value: $model.height,
+                    in: heightRange,
+                    step: 1
+                ) {
+                    Text("Observer Height")
+                } minimumValueLabel: {
+                    Text(heightRange.lowerBound, format: numberFormat)
+                } maximumValueLabel: {
+                    Text(heightRange.upperBound, format: numberFormat)
+                }
+                .listRowSeparator(.hidden, edges: .top)
+            }
+            .presentationDetents([.fraction(0.25)])
+            .navigationTitle("Settings")
+            .navigationBarTitleDisplayMode(.inline)
+            .toolbar {
+                ToolbarItem(placement: .confirmationAction) {
+                    Button("Done") {
+                        isPresented = false
+                    }
+                }
+            }
+        }
+    }
+}
+
+private extension ShowLineOfSightBetweenGeoelementsView {
+    /// View model responsible for setting up the 3D scene, graphics, animation,
+    /// and performing line of sight analysis between moving and static geoelements.
+    @MainActor
+    @Observable
+    final class Model {
+        /// Set of predefined waypoints for animating the taxi's movement.
+        private let points = [
+            Point(latitude: 40.748469, longitude: -73.984513),
+            Point(latitude: 40.747786, longitude: -73.985068),
+            Point(latitude: 40.747091, longitude: -73.983452),
+            Point(latitude: 40.747762, longitude: -73.982961)
+        ]
+        
+        /// Height of the observer in meters. Updates the observer graphic when changed.
+        var height = 20.0 {
+            didSet {
+                changeObserverHeight(height)
+            }
+        }
+        
+        private var frameIndex = 0
+        private let frameMax = 120
+        private var pointIndex = 0
+        
+        var error: Error?
```

Comment:
```suggestion
        /// The error shown in the error alert.
        @State private var error: Error?
```

This error should be on the view instead of the model because it is used in the view to present the error in the alert.  Please make `addGraphics()` throwing so the error can be handled in the view.

Reply:
This property should be moved to `ShowLineOfSightBetweenGeoelementsView` under `isPresented`, it does not need to be in the `ShowLineOfSightBetweenGeoelementsView.Model` class.

Reply:
Gotcha